pkg/distro/fedora/distro.go

```diff
@@ -700,6 +681,12 @@ func newDistro(version int) distro.Distro {
 	minimalrawZstdImgType.payloadPipelines = []string{"os", "image", "zstd"}
 	minimalrawZstdImgType.exports = []string{"zstd"}
 
+	// set iot services (these are version-dependent)
+	iotCommitImgType.defaultImageConfig.EnabledServices = iotServicesForVersion(&rd)
+	iotOCIImgType.defaultImageConfig.EnabledServices = iotServicesForVersion(&rd)
+	iotInstallerImgType.defaultImageConfig.EnabledServices = iotServicesForVersion(&rd)
+	iotSimplifiedInstallerImgType.defaultImageConfig.EnabledServices = iotServicesForVersion(&rd)
```

Comment:
Please, turn the image type definitions into a generator function and move the conditionals there. SImilar to https://github.com/osbuild/images/blob/9371af5db3bc8e8e9c4a620d29166a9149fdd35d/pkg/distro/rhel/rhel9/edge.go#L263-L299.

I don't like this approach, because the default image configuration is scattered in multiple places, which is not great for readability.

I'd like to consolidate this anyway once we move pkg definitions in RHEL to YAML and consolidate the distro interface implementation across all distros.

Reply:
I've rebased on `main` where we now have the `mk*` functions so the above code is gone.